src/cargo/sources/registry/mod.rs

```diff
@@ -850,9 +850,17 @@ impl<'cfg> Source for RegistrySource<'cfg> {
                 // names to the original name. The resolver will later
                 // reject any candidates that have the wrong name, and with this it'll
                 // along the way produce helpful "did you mean?" suggestions.
-                for name_permutation in
-                    index::UncanonicalizedIter::new(&dep.package_name()).take(1024)
+                // For now we only try the canonical lysing `-` to `_` and vice versa.
+                // More advanced fuzzy searching become in the future.
+                for name_permutation in [
+                    dep.package_name().replace('-', "_"),
+                    dep.package_name().replace('_', "-"),
+                ]
+                .into_iter()
                 {
```

Comment:
We don't need [`into_iter` for array](https://doc.rust-lang.org/edition-guide/rust-2021/IntoIterator-for-arrays.html) anymore :)



```suggestion
                ] {
```